coordinator/coordinator.go

```diff
@@ -823,54 +792,106 @@ func (c *Coordinator) runFeatureMaterializeJob(resID metadata.ResourceID, schedu
 		return fmt.Errorf("materialize feature register: %v", err)
 	}
 	c.Logger.Debugw("Resource Table Created", "id", featID, "schema", schema)
-	needsOnlineMaterialization := strings.Split(string(featureProvider.Type()), "_")[1] == "ONLINE"
-	if needsOnlineMaterialization {
-		c.Logger.Info("Starting Materialize")
-		jobRunner, err := c.Spawner.GetJobRunner(runner.MATERIALIZE, serialized, resID)
-		if err != nil {
-			return fmt.Errorf("could not use %s as online store: %w", featureProvider.Name(), err)
-		}
-		completionWatcher, err := jobRunner.Run()
-		if err != nil {
-			return fmt.Errorf("failed to run job: %w", err)
-		}
-		if err := completionWatcher.Wait(); err != nil {
-			return fmt.Errorf("failed to complete job: %w", err)
+
+	materializedRunnerConfig := runner.MaterializedRunnerConfig{
+		OnlineType:    pt.Type(featureProvider.Type()),
+		OfflineType:   pt.Type(sourceProvider.Type()),
+		OnlineConfig:  featureProvider.SerializedConfig(),
+		OfflineConfig: sourceProvider.SerializedConfig(),
+		ResourceID:    provider.ResourceID{Name: resID.Name, Variant: resID.Variant, Type: provider.Feature},
+		VType:         provider.ValueTypeJSONWrapper{ValueType: vType},
+		Cloud:         runner.LocalMaterializeRunner,
+		IsUpdate:      false,
+	}
+
+	isImportToS3Enabled := false
+	if featureProvider.Type() == string(pt.DynamoDBOnline) {
```

Comment:
could probably pull this out into a function:
`isImportToS3Enabled, err := c.s3ImportCheck(featureProvider)`